src/main/java/io/divolte/server/filesinks/hdfs/HdfsFileManager.java

```diff
@@ -0,0 +1,223 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.divolte.server.filesinks.hdfs;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Objects;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.mortbay.log.Log;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+
+import io.divolte.server.AvroRecordBuffer;
+import io.divolte.server.config.FileSinkConfiguration;
+import io.divolte.server.config.HdfsSinkConfiguration;
+import io.divolte.server.config.ValidatedConfiguration;
+import io.divolte.server.filesinks.FileManager;
+
+@ParametersAreNonnullByDefault
+public class HdfsFileManager implements FileManager {
+    private static final Logger logger = LoggerFactory.getLogger(HdfsFileManager.class);
+
+    private final static String INFLIGHT_EXTENSION = ".partial";
+
+    private final FileSystem hdfs;
+    private final short replication;
+    private final String workDir;
+    private final String publishDir;
+    private final Schema writeSchema;
+
+    private HdfsFileManager(final FileSystem hdfs, final short replication, final String workDir, final String publishDir, final Schema schema) {
+        this.hdfs = Objects.requireNonNull(hdfs);
+        this.workDir = Objects.requireNonNull(workDir);
+        this.publishDir = Objects.requireNonNull(publishDir);
+        this.writeSchema = Objects.requireNonNull(schema);
+        this.replication = replication;
+    }
+
+    @Override
+    public DivolteFile createFile(final String name) throws IOException {
+        return new HdfsDivolteFile(name);
+    }
+
+    public static FileManagerFactory newFactory(final ValidatedConfiguration configuration, final String sinkName, final Schema schema) {
+        return new HdfsFileManagerFactory(configuration, sinkName, schema);
+    }
+
+    public class HdfsDivolteFile implements DivolteFile {
+        private final FSDataOutputStream stream;
+        private final DataFileWriter<GenericRecord> writer;
+        private final Path inflightPath;
+        private final Path publishPath;
+
+        @SuppressWarnings("resource")
```

Comment:
It's about the unassigned `Closeable` (`new GenericDatumWriter<>(writeSchema)`). We hold no direct reference to it, so the warning machine thinks we'll never close it.